editor/src/components/canvas/commands/insert-element-insertion-subject.ts

```diff
@@ -59,10 +59,19 @@ export const runInsertElementInsertionSubject: CommandFunction<InsertElementInse
         return
       }
 
-      const insertionResult = insertElementAtPath_DEPRECATED(
+      const insertionPath = getDefaultInsertionPathForElementPath(
+        targetParent,
         editor.projectContents,
-        underlyingFilePath,
-        childInsertionPath(targetParent),
+        editor.nodeModules.files,
+        editor.canvas.openFile?.filename,
+        editor.jsxMetadata,
+      )
+      if (insertionPath == null) {
+        return // maybe this should throw instead?
```

Comment:
Definitely not. That would mean the editor throws an exception when you try to insert into a non-empty clause.

Reply:
I'm not sure where this action is used from, but let me note that this way we don't have automatic wrapping to a fragment when inserting into a non-empty clause, it will just silently do nothing.
I guess depending on the actual feature we should either
(1) show a toast
(2) implement fragment insertion with a new type of InsertionPath
(3) just leave it as it is

If it turns out that this function is used in all different contexts and this can not be decided, I would change the action object to store the InsertionPath directly, and then the call sites can decide how to call the action creator.

Reply:
> I would change the action object to store the InsertionPath directly, and then the call sites can decide how to call the action creator.

in general I think this is very important to keep this in mind! 
this is a very fuzzily defined task, but we are basically looking for the "highest" place where we can put the slot-like behavior. the ideal is to put it in code that "talks" about behavior / ux. so in this case, that should be the strategies, and not the commands. the commands should be low-level executors, not place where behavior lives.

reality is way fuzzier than this, and there's sometimes not a good solution. but something to keep in mind

Reply:
I think that yeah, providing the insertion path in the strategies (which use this one) would be the preferable solution. Will adjust